src/fosslight_util/download.py

```
@@ -576,6 +576,55 @@ def decide_checkout(checkout_to="", tag="", branch="", git_url=""):
     return "", ""
 
 
+def _git_url_with_http_credentials(git_url, id="", git_token=""):
```

Ruff (0.15.20)

[error] Function argument `id` is shadowing a Python builtin

(A002)